x-pack/libbeat/cmd/enroll.go

```diff
@@ -17,6 +18,10 @@ import (
 	"github.com/elastic/beats/x-pack/libbeat/management/api"
 )
 
+var (
+	metadata = common.SettingFlag(nil, "meta", "Metadata configuration specified as K/V: K=V")
+)
```

Comment:
Let's remove this special flags for now, I prefer that either people define them in the YAML or use our generic syntax to overrides items in the YAML like this.
```
-E "management.metadata.hello=world"
```

IIRC we don't want to support multiple levels of keys when creating metadata, so we should flatten the values before sending them to ES. I guess that operation is needed on both side? 

```json
// Multiple levels of keys
{
  "hello": {
    "world": "more hello"
  }
}
//flatten
{
"hello.world": "more hello"
}
```

@mattapperson I know that we do not want to allow people to edit keys push from beats to the API, I think we should automatically prefix them with "beat", "remote" or something else, if we do this we could assume that if the update is coming from the beat itself we can always update the data.

Reply:
sounds good to me. `meta` tag is a nice way how to get rid of overhead with typing `management.metadata` all over again. 
but i'm ok with not having it for now
about flattening, in the issue we talked that that metadata should be of simple type (string/number) or array of simple types. So is it necessary?
I can add type check and throw if a user specifies a nested type